server/src/main/java/org/spine3/server/aggregate/AggregatePartRepositoryLookup.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2017, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.server.aggregate;
+
+import com.google.common.base.Optional;
+import com.google.protobuf.Message;
+import org.spine3.server.BoundedContext;
+
+/**
+ * A helper class for finding and checking {@code AggregatePartRepository}.
+ *
+ * <p>This class is used by {@link AggregateRoot} to find repositories for its parts.
+ *
+ * @param <I> the type of the IDs of the repository to find
+ * @param <S> the type of the state of aggregate parts of managed by the target repository
+ * @author Alexander Yevsyukov
+ */
+class AggregatePartRepositoryLookup<I, S extends Message> {
```

Comment:
I thought that being package-access it doesn't need the annotation. It is internal, though.